scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,167 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+def gttrf_to_lu(n, dtype, dlf, df, duf, du2f, ipiv):
+    U = np.diag(df, 0) + np.diag(duf, 1) + np.diag(du2f, 2)
+    L = np.eye(n, dtype=dtype)
+    for i, m in enumerate(dlf):
+        # L is given in a factored form.
+        # See http://www.hpcavf.uclan.ac.uk/softwaredoc/sgi_scsl_html/sgi_html/ch03.html
+        piv = ipiv[i] - 1
+        # right multiply by permutation matrix
+        L[:, [i, piv]] = L[:, [piv, i]]
+        # right multiply by Li, rank-one modification of identity
+        L[:, i] += L[:, i+1]*m
+
+    # one last permutation
+    i, piv = -1, ipiv[-1] - 1
+    # right multiply by final permutation matrix
+    L[:, [i, piv]] = L[:, [piv, i]]
+    return (L, U)
+
+
+@pytest.mark.parametrize("dtype,trans", list(zip(DTYPES, ["N"]*len(DTYPES)))
+                         + list(zip(REAL_DTYPES, ["T"]*len(REAL_DTYPES)))
+                         + list(zip(COMPLEX_DTYPES,
+                                    ["C"]*len(COMPLEX_DTYPES))))
+@pytest.mark.parametrize("fact,dl_d_du_lambda",
+                         [("F", lambda dl, d, du: get_lapack_funcs('gttrf',
+                                                                   dtype=d.dtype)(dl, d, du)),
+                          ("N", lambda dl, d, du: (None, None, None, None,
+                                                   None, None))])
+def test_gtsvx(dtype, trans, fact, dl_d_du_lambda):
+    """
+    This test uses ?gtsvx to solve a random Ax=b system for each dtype.
+    It tests that the outputs define an LU matrix, that inputs are unmodified,
+    transposal options, incompatible shapes, singular matrices, and
+    singular factorizations. It parametrizes DTYPES and the fact value along
```

Comment:
```suggestion
    singular factorizations. It parametrizes DTYPES and the `fact` value along
```
I would do something to indicate that "fact" is the name of a variable, not some jargon that the reader doesn't understand.